SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -298,10 +354,90 @@ void UGlobalStateManager::AuthorityChanged(bool bWorkerAuthority, Worker_EntityI
 	{
 		// Make sure we update our known entity id for the GSM when we receive authority.
 		GlobalStateManagerEntityId = CurrentEntityID;
+
+		SetCanBeginPlay(true);
+		BecomeAuthoritativeOverAllActors();
+		TriggerBeginPlay();
+
+		// Start accepting players only AFTER we've triggered BeginPlay
 		SetAcceptingPlayers(true);
 	}
 }
 
+void UGlobalStateManager::BeginDestroy()
+{
+	Super::BeginDestroy();
+
+#ifdef WITH_EDITOR
+	if (NetDriver->StaticComponentView->HasAuthority(GlobalStateManagerEntityId, SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID))
```

Comment:
I saw instances where this NetDriver was nullptr. It looked like it was destroying the CDO though, so think it's safe to just througha nullptr check in.

Reply:
Also use UGlobalStateManager::HasAuthority()